dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/security/DefaultAnalyticsSecurityManager.java

```diff
@@ -96,6 +99,28 @@ public DefaultAnalyticsSecurityManager( DataApprovalLevelService approvalLevelSe
     // AnalyticsSecurityManager implementation
     // -------------------------------------------------------------------------
 
+    /**
+     * Will remove/exclude, from DataQueryParams, any category option that the
+     * current user is not allowed to read.
+     *
+     * @param programCategories the categories related to this program.
+     */
+    void excludeNonAuthorizedCategoryOptions( final List<Category> programCategories )
```

Comment:
Add modifier (private)

Reply:
We decided to leave as is to favour testability.